### ISSUE TITLE
What if I-X is predicted at the start of the sentence?

### ISSUE THREAD

**Issue by ZenBel** [DandyQi/MaskedCRF#4]
Hi and thanks for the nice work on masked CRF. I have implemented your model in pytorch and noticed an improvement in f1-score for the dataset I am using. However, I noticed that the model still predicts I-X tags at the beginning of the sentence. An example of this behaviour would be:

"London is a city" is predicted as "I-CITY O O O", while the correct prediction should be "B-CITY O O O".

Any idea, how to extend the masked CRF to cope with this kind of behaviour?

Thanks in advance,

Zeno

**Comment by TianwenWei** [comment]
Hi ZenBel. I have never come across such behaviour. One solution for this would be to add a special `<SOS>` token at the beginning of each sentence, and that `<SOS>` should always be labeled as an `O`. At inference time, "London is a city" becomes "`<SOS>` London is a city", and the model should be able to predict the `<SOS>` token as an `O`. Then by masking illegal transitions, `London` will become a `B-city` instead of an `I-city`.

Regards

**Comment by ZenBel** [comment]
Yes, adding a `<sos>` token at the beginning of the sentence fixes it. Thanks for the reply!
